### PR TITLE
Add saydoku to Saylua

### PR DESCRIPTION
**Pull request by tiffz**
Currently you can't actually submit the puzzle to get a prize, but the UI is essentially done, there is clientside validation of your puzzle, and puzzles which have a solution are generated. 
screenshot 2017-02-27 23 44 50

### CODE REVIEW

**Comment by tiffz** [comment]
Additional screenshot to show what the error highlighting looks like (also what inputs look like). 
screenshot 2017-02-27 23 46 40

File: saylua/modules/home/static-source/es6/NewsPuzzle/Main.js [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+  submit.addEventListener("click", function () {
+    // Note this is 2D data (9x9) that is flattened into a 1D array (1x81)
+    let numbers = puzzle.getElementsByClassName("puzzle-number");
+    let emptyCount = 0;
+    let invalidCount = 0;
+
+    let rowCollisions = {},
+      colCollisions = {},
+      boxCollisions = {};
```
Comment (NoiSek) [inline saylua/modules/home/static-source/es6/NewsPuzzle/Main.js]: something something `\n`

**Review by NoiSek — APPROVED** [review]
Neat.

File: saylua/modules/home/static-source/es6/NewsPuzzle/Main.js (hunk 1 above)
Comment (tiffz) [inline saylua/modules/home/static-source/es6/NewsPuzzle/Main.js]: i feel like i'm missing a few l's :(

**Pull request merged**